packages/core/src/artifacts.ts

```diff
@@ -268,23 +269,24 @@ export const convertEthersTransactionReceipt = (
 }
 
 /**
- * Makes contract deployment artifacts on a single network for a single deployment.
+ * Makes contract deployment artifacts on a single network for a single deployment. Mutates the
+ * input `artifacts` object.
```

Comment:
The easiest way to implement this patch was to mutate the previous artifacts (i.e. `artifacts`)